src/units/abilities.cpp

```diff
@@ -2381,7 +2382,17 @@ effect::effect(const unit_ability_list& list, int def, const_attack_ptr att, EFF
 		effect_list_.push_back(val.second);
 	}
 
-	composite_value_ = static_cast<int>((value_set + addition) * multiplier / divisor);
+	/* Using 'substraction' independant of 'addition' for abilities with same id and 'add' and 'sub'
+	 * for what rule of calculation was base+ add_value - sub_value
+	 * instead of base + (add_value or -sub_value).
```

Comment:
The "instead of" makes sense in the context of a PR (because we know what's changing), but won't make sense when looking at this code in future.

"Additional and subtraction are independent since Wesnoth 1.19.4. Prior to that, they affected each other."

Reply:
Fixed